src/parse.rs

```diff
@@ -16,3 +16,38 @@ impl Parse for AsyncItem {
         Ok(AsyncItem(item))
     }
 }
+
+#[derive(Default)]
+pub struct RecursionArgs {
+    pub send_bound: bool,
+}
+
+impl Parse for RecursionArgs {
+    fn parse(input: ParseStream) -> Result<Self> {
+        // Start with default args
+        let mut args = RecursionArgs::default();
+
+        // Parse the provided args ( we are looking for whether or not `?Send` is specified )
+        let lookahead = input.lookahead1();
+        if lookahead.peek(Token![?]) {
+            input.parse::<Question>()?;
+            let lookahead = input.lookahead1();
+            if lookahead.peek(Ident) {
+                let ident: Ident = input.parse()?;
+                if ident.to_string() == "Send" {
+                    args.send_bound = false;
+                } else {
+                    return Err(input.error("expected `?Send` or empty"));
+                }
+            } else {
+                return Err(lookahead.error());
+            }
+        } else if input.is_empty() {
+            args.send_bound = true;
+        } else {
+            return Err(lookahead.error());
+        }
+
+        Ok(args)
+    }
+}
```

Comment:
I think this could be simplified a bit using [custom keywords](https://docs.rs/syn/1.0.17/syn/macro.custom_keyword.html) - I haven't checked if this works, but something like:

```suggestion
mod kw {
    syn::custom_keyword!(Send);
}

impl Parse for RecursionArgs {
    fn parse(input: ParseStream) -> Result<Self> {
        if input.peek(Token![?]) {
            input.parse::<Token![?]>()?;
            input.parse::<kw::Send>()?;
            Ok(Self { send_bound: false })
        } else {
            Ok(Self::default())
        }
    }
}
```

Reply:
Custom keywords are perfect! Fixed.